scripts/zones/Ilrusi_Atoll/mobs/Undead_Leech.lua

```diff
@@ -2,15 +2,19 @@
 -- Area: Ilrusi Atoll (Extermination)
 --  Mob: Undead Leech
 -----------------------------------
+require("scripts/globals/assault")
+-----------------------------------
 local entity = {}
 
-entity.onMobDeath = function(mob, player, isKiller)
+entity.onMobSpawn = function(mob)
+    assaultUtil.adjustMobLevel(mob, mob:getID())
 end
 
-entity.onMobDespawn = function(mob)
-    local instance = mob:getInstance()
-
-    instance:setProgress(instance:getProgress() + 1)
+entity.onMobDeath = function(mob, player, isKiller, firstCall)
```

Comment:
Not sure about firstCall in this context.  Fourth parameter on our end looks like isWeaponskillKill, but might be worth confirming that is the intended value.

Reply:
I think eventually we're going to have to change the function signature for this to be:
```
onMobDeath(mob, player, param)

params = {
    isKiller
    firstCall
    isWSKill
    isMagicKill
    etc.
}
```

or something like this

Reply:
Teo just wants to say Teo is cool with using a params table input there. In the past envisioned a typemask for killtypes and didn't try to do it because of the obvious difficulties in making that happen and using it. Teo will now cease 3rd persons peak and attempt "sleep" which is allegedly a thing normal people do on the regular.